To: vim_dev@googlegroups.com
Subject: Patch 7.3.185
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.185
Problem:    ":windo g/pattern/q" closes windows and reports "N more lines".
	    (Tim Chase)
Solution:   Remember what buffer ":global" started in. (Jean-Rene David)
Files:	    src/ex_cmds.c


*** ../vim-7.3.184/src/ex_cmds.c	2011-04-11 21:35:03.000000000 +0200
--- src/ex_cmds.c	2011-05-10 15:58:48.000000000 +0200
***************
*** 5365,5372 ****
  global_exe(cmd)
      char_u	*cmd;
  {
!     linenr_T	old_lcount;	/* b_ml.ml_line_count before the command */
!     linenr_T	lnum;		/* line number according to old situation */
  
      /*
       * Set current position only once for a global command.
--- 5365,5373 ----
  global_exe(cmd)
      char_u	*cmd;
  {
!     linenr_T old_lcount;	/* b_ml.ml_line_count before the command */
!     buf_T    *old_buf = curbuf;	/* remember what buffer we started in */
!     linenr_T lnum;		/* line number according to old situation */
  
      /*
       * Set current position only once for a global command.
***************
*** 5410,5417 ****
  	msg_didout = FALSE;
  
      /* If substitutes done, report number of substitutes, otherwise report
!      * number of extra or deleted lines. */
!     if (!do_sub_msg(FALSE))
  	msgmore(curbuf->b_ml.ml_line_count - old_lcount);
  }
  
--- 5411,5420 ----
  	msg_didout = FALSE;
  
      /* If substitutes done, report number of substitutes, otherwise report
!      * number of extra or deleted lines.
!      * Don't report extra or deleted lines in the edge case where the buffer
!      * we are in after execution is different from the buffer we started in. */
!     if (!do_sub_msg(FALSE) && curbuf == old_buf)
  	msgmore(curbuf->b_ml.ml_line_count - old_lcount);
  }
  
*** ../vim-7.3.184/src/version.c	2011-05-10 15:52:10.000000000 +0200
--- src/version.c	2011-05-10 15:57:40.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     185,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
79. All of your most erotic dreams have a scrollbar at the right side.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
